File: challenges/374/python/solution.py
#!/usr/bin/env python
from challenges.utils import expect


# A dummy placeholder just to avoid errors highlighting in the IDE
def guess(n: int) -> int:
    return n


class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 1, n
        while True:
            middle = (left + right) // 2
            guess_result = guess(middle)
            if guess_result == -1:
                right = middle - 1
            elif guess_result == 1:
                left = middle + 1
            else:
                return middle


if __name__ == "__main__":
    pass

File: challenges/374/js/solution.ts
/**
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */

export const guess = (num: number): number => 1;

const guessNumber = (n: number): number => {
  let left = 1,
    right = n;
  while (true) {
    const middle = Math.floor((left + right) / 2);
    const guessResult = guess(middle);
    if (guessResult === 1) {
      left = middle + 1;
    } else if (guessResult === -1) {
      right = middle - 1;
    } else {
      return middle;
    }
  }
};
